modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,420 @@
-      end
-      tl.first.join
-      tl.delete_if { |t| not t.alive? }
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::MX
+      records << "#{r.exchange}"
+      report_host(host: r.exchange, name: domain, info: 'MX')
+      report_service(host: r.exchange, name: 'smtp', port: 25, proto: 'tcp')
+      vprint_good("#{domain}: MX: #{r.exchange}")
     end
+    return if records.none?
+    path = store_loot(
+      'ENUM_MX',
+      'text/plain',
+      domain,
+      records,
+      'ENUM_MX')
+    print_good('Saved file to: ' + path)
+    records
   end
 
-  # SRV Record Enumeration
-  def srvqry(dom,nssrv)
-    print_status("Enumerating SRV records for #{dom}")
-    i, a = 0, []
-    # Most common SRV Records
-    srvrcd = [
-      "_gc._tcp.","_kerberos._tcp.", "_kerberos._udp.","_ldap._tcp.","_test._tcp.",
-      "_sips._tcp.","_sip._udp.","_sip._tcp.","_aix._tcp.","_aix._tcp.","_finger._tcp.",
-      "_ftp._tcp.","_http._tcp.","_nntp._tcp.","_telnet._tcp.","_whois._tcp.","_h323cs._tcp.",
-      "_h323cs._udp.","_h323be._tcp.","_h323be._udp.","_h323ls._tcp.","_h323ls._udp.",
-      "_sipinternal._tcp.","_sipinternaltls._tcp.","_sip._tls.","_sipfederationtls._tcp.",
-      "_jabber._tcp.","_xmpp-server._tcp.","_xmpp-client._tcp.","_imap._tcp.","_certificates._tcp.",
-      "_crls._tcp.","_pgpkeys._tcp.","_pgprevokations._tcp.","_cmp._tcp.","_svcp._tcp.","_crl._tcp.",
-      "_ocsp._tcp.","_PKIXREP._tcp.","_smtp._tcp.","_hkp._tcp.","_hkps._tcp.","_jabber._udp.",
-      "_xmpp-server._udp.","_xmpp-client._udp.","_jabber-client._tcp.","_jabber-client._udp."]
-    srvrcd.each do |srvt|
-      trg = "#{srvt}#{dom}"
-      query = @res.query(trg , Net::DNS::SRV)
-      next unless query
-      query.answer.each do |srv|
-        next if srv.type == "CNAME"
-        print_status("SRV Record: #{trg} Host: #{srv.host} Port: #{srv.port} Priority: #{srv.priority}")
-      end
+  def get_soa(domain)
+    resp = dns_query(domain, 'SOA')
+    return if resp.blank? || resp.answer.blank?
+
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::SOA
+      records << r.mname
+      report_host(host: r.mname, info: 'SOA')
+      vprint_good("#{domain}: SOA: #{r.mname}")
     end
-    print_status("Done")
+    return if records.none?
+    path = store_loot(
+      'ENUM_SOA',
+      'text/plain',
+      domain,
+      records,
+      'ENUM_SOA')
+    print_good('Saved file to: ' + path)
+    records
   end
 
-  # For Performing Zone Transfers
-  def axfr(target, nssrv)
-    print_status("Performing zone transfer against all nameservers in #{target}")
-    if not nssrv.nil?
-      @res.nameserver=(nssrv)
-      @nsinuse = nssrv
+  def get_txt(domain)
+    resp = dns_query(domain, 'TXT')
+    return if resp.blank? || resp.answer.blank?
+
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::TXT
+      records << r.txt
+      report_service(
```

Comment:
I would suggest removing all/most of the calls to `report_service` in the `get_*` methods.  In most cases, you are not directly or even indirectly querying the host that is the NS for `domain`, but rather something else entirely.  IOW, unless  `NS` == `DOMAIN`, you should not `report_service` here.